medusajs/nextjs-starter-medusa
# feat: add product module

**In this PR:**
- Created product module endpoints in `src/app/api` for:
  - Fetch a product by handle
  - List all products, w/ pagination
  - List all products in a collection, based on collection id, w/ pagination
  - List all collections, w/ pagination
  - Fetch a collections metadata and products by handle
- Added the `productModule` feature flag in `store.config.json`
- Refactored `src/lib/data/index.ts` to contain all relevant product and collection fetching functions. Within the functions, either the standard Medusa API or the Product Module will be called, depending on the feature flag setting.

**To test locally:**
- Clone, yarn and `yarn dev` this project. Be sure to have a Medusa server running on port 9000.
- Test the store on http://localhost:8000/. By default it will use the standard API.
- After testing, change the `productModule` feature flag in `store.config.json` to `true`.
- Restart the server. Now test again on http://localhost:8000/. All product fetches should now come from the product module endpoints.
